### ISSUE TITLE
uninitialized constant Puppet::Provider::CommandDefiner

### ISSUE THREAD

**Issue by cova-fe** [fiddyspence/puppet-sysctl#19]
Hi all, I've installed this module (really useful indeed!) but I'm unable to make it work on machines with puppet agent < 3.X 
On machines with 2.7 or 2.6 (I have some legacy issues here :) )I got this error:

err: Could not retrieve catalog from remote server: Could not intern from pson: Could not autoload sysctl: Could not autoload /var/lib/puppet/lib/puppet/provider/sysctl/linux3.rb: uninitialized constant Puppet::Provider::CommandDefiner

Basically it seems that linux3.rb gets loaded even on 2.X agents (the server is 3.4 in either cases)

I searched but found no indication of versions supported by this module; moreover I found this scary warning, trying to look at the code:

"This class is part of a private API. You should avoid using this class if possible, as it may be removed or be changed in the future."

so I stopped digging and opened an issue :)
